Global styles: improve navigation logic for revisions screen

ciampo opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->

Part of https://github.com/WordPress/gutenberg/issues/65794

Tweak the code in the global styles sidebar, improving the logic around navigating to and away from the revisions screen.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

Navigating to/from the revisions screen wasn't always triggering the correct animation, and sometimes the exit animation was playing twice.

> [!NOTE]
> Navigating _to_ the revisions screen can still trigger a double animation, but I believe this is caused by how the screen itself handles data fetching updates via a mix of data store selectors and internal state, which causes the component to re-render and potentially re-trigger the CSS entry animation. This PR doesn't aim at improving that aspect — to improve it, we should probably refactor the revisions screen's internal code more substantially, similarly to the work that @jsnajdr did in https://github.com/WordPress/gutenberg/pull/65564

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

Since the button that shows/hides the screen is actually outside of `Navigator`, it can't use `Navigator.Button` or `useNavigator` directly. Instead, interacting with the button changes a value in the data store, and then code in the global styles sidebar react to those changes to programmatically navigate to a new screen.

This PR removes any similar logic in the revisions screen itself and moves any needed logic to the file hosting the root `Navigator` component, next to where other similar logic is.

This PR also add the `isBack` flag when navigating to the home screen, which makes the animation more aligned with the user's expectations.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->

- In the site editor, open the global styles sidebar;
- Interact with the revisions screen:
  - Open and close the screen by clicking on the icon button in the global styles sidebar header;
    - [ ] When toggling the revisions screen _off_, the screen leaves with a backwards animation (on `trunk`, it navigates forward)
  - Navigate back to the home screen by clicking the back button in the revisions screen's header;
- As explained above, the double navigation that can happen when navigating _to_ the revisions screen already exists on `trunk`, and fixing it is out of the scope of this PR.

## Screenshots or screencast <!-- if applicable -->

| Before (trunk) | After (this PR) |
|---|---|
|  <video src="https://github.com/user-attachments/assets/7de59332-0b88-4241-b111-016f798c25df" /> | <video src="https://github.com/user-attachments/assets/14f371fc-be0f-4082-94b0-05ed8b3d6968" /> |

ciampo committed:
Global styles: improve navigation logic for revisions screen

tyxla commented on packages/edit-site/src/components/global-styles/ui.js:
```diff
@@ -267,7 +269,14 @@ function GlobalStylesEditorCanvasContainerLink() {
-					goTo( '/' );
+					goTo( '/', { isBack: true } );
+				}
+				break;
```
Should we just let the `case` fall through to the `default` one?

ciampo replied:
https://github.com/WordPress/gutenberg/pull/65946/commits/1d52e6984b550bdb188811bb097fac48a4c728e3

tyxla replied:
You could also keep the comment if you find it useful:

```js
case 'style-book':
	/*
	 * The stand-alone style book is open
	 * and the revisions panel is open,
	 * close the revisions panel.
	 * Otherwise keep the style book open while
	 * browsing global styles panel.
	 *
	 * Falling through as it matches the default scenario.
	 */
default:
	// In general, if the revision screen is in view but the
	// `editorCanvasContainerView` is not a revision view, close it.
	// This also includes the scenario when the stand-alone style
	// book is open, in which case we want the user to close the
	// revisions screen and browse global styles.
	if ( isRevisionsOpen ) {
		goTo( '/', { isBack: true } );
	}
```

I'm fine with omitting it or keeping it, whatever you prefer.

tyxla commented on packages/edit-site/src/components/global-styles/screen-revisions/index.js:
```diff
@@ -71,6 +69,8 @@ function ScreenRevisions() {
 		currentEditorGlobalStyles
 	);
 
+	// The actual code that triggers the revisions screen to navigate back
+	// to the home screen in in `packages/edit-site/src/components/global-styles/ui.js`.
```
Thanks for the comment 👍

ciampo committed:
Merge style book case into default, add comment

ciampo committed:
Remove unnecessary goTo calls

ciampo commented on packages/edit-site/src/components/global-styles-sidebar/index.js:
All of the `goTo` calls in this file are no-ops, because `useNavigator` is being called in a component that is not a child of a `Navigator` component.

ciampo commented:
As i was addressing some feedback, I noticed a few places where calling `goTo` didn't make any difference:
- in some cases, `goTo` was effectively a no-op ()
- in general, it looks we're already reacting to changes top the `editorCanvasContainerView` and call `goTo` consequently

I confess that I'm a bit loss with respect to what is the expected behaviour of all possible interactions between global styles, revisions, style book, and additional CSS styles. Maybe @ramonjd @andrewserong @jameskoster can help understand this expected behaviour, so that we can clean up the logic?

jameskoster commented:
The Styles panel was originally built with a narrow scope and over time has become a bit of a Frankenstein's monster. The whole iA needs a rethink (https://github.com/WordPress/gutenberg/issues/47369), and until that's done I'm hesitant to give any suggestions that may make the monster larger. 

There's some related work in https://github.com/WordPress/gutenberg/pull/65619, demonstrating this is a big project that needs careful thought.

Sorry for not being much help 🙈

ciampo commented:
Got it. Maybe I'll try to use my best common sense, and we can hopefully merge this PR as it cleans up navigator-related code quite a bit

tyxla commented on packages/edit-site/src/components/global-styles-sidebar/index.js:
```diff
@@ -86,21 +86,17 @@ export default function GlobalStylesSidebar() {
 	}, [ shouldClearCanvasContainerView ] );
 
 	const { setIsListViewOpened } = useDispatch( editorStore );
-	const { goTo } = useNavigator();
```
Great catch 👏 

FWIW we should also remove the `useNavigator` import.

ramonjd replied:
I just did this while testing. 👍🏻

ramonjd committed:
Remove useNavigator import

ramonjd reviewed (approved):
Thank you for improving this!

I cannot break it. Revisions navigation works as expected for me.

https://github.com/user-attachments/assets/acbee15f-12d2-4224-be92-50959dd39049

ciampo committed:
Apply feedback
